system/datatypes/peer_reviewed_article.php

```diff
@@ -778,6 +799,153 @@ protected function discussionClosedAction(HyphaRequest $request) {
 		return ['redirect', $this->constructFullPath($this->pagename)];
 	}
 
+	/**
+	 * Constructs the WymHTMLForm for the comment form.
+	 *
+	 * @param HyphaDomElement $comment
+	 * @param array $values
+	 * @return HTMLForm
+	 */
+	protected function createCommentModerateForm($comment, $values = []) {
+		$html = <<<EOF
+			<div>
+				<strong><label for="[[field-name-reason]]">[[reason]]</label></strong>
+				<br/>
+				<input type="text" name="[[field-name-reason]]" id="[[field-name-reason]]"></input>
+			</div>
+EOF;
+
+		$vars = [
+			'reason' => __('art-moderate-reason'),
+			'field-name-reason' => self::FIELD_NAME_MODERATE_REASON,
+		];
+
+		$html = hypha_substitute($html, $vars);
+		return new HTMLForm($html, $values);
+	}
+
+	/**
+	 * Allows editing the moderation status of a public comment.
+	 *
+	 * @param HyphaRequest $request
+	 * @return array|null
+	 */
+	protected function commentModerateView(HyphaRequest $request) {
+		if (!$this->O_O->isUser()) {
+			notify('error', __('login-to-perform-action'));
+			return ['redirect', $this->constructFullPath($this->pagename)];
+		}
+
+		/** @var HyphaDomElement|null $comment */
+		$commentId = $request->getArg(1);
+		$comment = $this->xml->document()->getElementById($commentId);
+		if (!$comment instanceof HyphaDomElement || $comment->tagName != self::FIELD_NAME_DISCUSSION_COMMENT) {
+			$this->xml->unlock();
+			notify('error', __('not-found')); // 404
+		}
+
+		$values = [
+			self::FIELD_NAME_MODERATE_REASON => $comment->getAttribute(self::FIELD_NAME_DISCUSSION_MODERATION_REASON),
+		];
+		$form = $this->createCommentModerateForm($comment, $values);
+
+		return $this->commentModerateViewRender($request, $form, $comment);
+	}
+
+	/**
+	 * Appends moderate form to #main and adds buttons
+	 *
+	 * @return null
+	 */
+	protected function commentModerateViewRender(HyphaRequest $request, HTMLForm $form, HyphaDomElement $comment) {
+		// update the form dom so that error can be displayed, if there are any
+		$form->updateDom();
+		$hidden = $comment->getAttribute(self::FIELD_NAME_DISCUSSION_MODERATION_STATUS) === self::MODERATION_STATUS_HIDDEN;
+
+		$main = $this->html->find('#main')->first();
+		$commentDiv = $this->html->createElement('div')->appendTo($main);
+		$commentDiv->addClass('moderation-comment');
+
+		$prefix = $this->html->createElement('span')->appendTo($commentDiv);
+		$prefix->addClass('prefix');
+		if ($hidden) {
+			$prefix->setText(__('art-comment-is-hidden'));
+		} else {
+			$prefix->setText(__('art-comment-to-moderate'));
+		}
+
+		$this->createCommentDomElement($comment, false, false)->appendTo($commentDiv);
+
+		$main->append($form);
+
+		$commands = $this->html->find('#pageEndCommands');
+		$commands->append($this->makeActionButton(__('cancel')));
+
+		$path = str_replace('{id}', $comment->getId(), self::PATH_COMMENTS_MODERATE);
+		$hide_title = $hidden ? __('art-moderate-change-reason') : __('art-moderate-hide-comment');
```

Comment:
the name of this variable is a bit confusion, it's not always hidden.

Reply:
Do you mean that "hidden" is confusing in general, because you can still show the comment by expanding it? This is true, but I couldn't think of any better name ("hidden-by-default" is too complicated, I think). And it *is* hidden, it's just also
possible to reveal what's hidden :-)

Or do you mean that this particular button does not always hide the comment, it sometimes only changes the reason? That's not really what happens in practice, though, since this button always uses the HIDE command, so it always sets the moderate-status to hidden and sets the moderate-reason. So if the status is already hidden, this (in the view of the user) only changes the reason, but underwater it still does both.

Maybe this clarifies things? If not, let me know what you mean exactly, and maybe you also have a suggestion to improve?

Reply:
I think I misread the `$hidden = $comment->getAttribute(self::FIELD_NAME_DISCUSSION_MODERATION_STATUS) === self::MODERATION_STATUS_HIDDEN;` for `$hidden = $comment->getAttribute(self::FIELD_NAME_DISCUSSION_MODERATION_STATUS);`.
My bad.